### PR TITLE
Update the method determining whether to installation is fresh or not by taking into account empty database

### PR DESCRIPTION
**Pull request by ssougnez**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/3753

### Description

Following the feature request aforementioned, I made these two minor changes to allow Umbraco to get installed when a connection string referring to an empty database is set.

Basically, I made two changes:

- "SaveConnectionString" in "DatabaseContext" is now internal.
- The property "IsBrandNewInstall" of "InstallHelper" now checks whether a connection string has been defined and if this one points to an empty database. If it's the case, the version and connection string are cleaned from the web.config to force a brand new installation.

### CODE REVIEW

**Comment by emmaburstow** [comment]
Hi @ssougnez 

Not sure why we've neglected to thank you for this but here it is! Thanks! We'll take a look over and let you know if we need anything.

Emma

**Comment by nul800sebastiaan** [comment]
Hey there @ssougnez !

I'm writing today after announcing at [Codegarden](https://codegarden19.com/), our yearly conference, that the upcoming version 7.15.0 will be the last Umbraco v7 release with new features in it. From now on all our efforts will be focused on version 8 instead. 🎱

We're wrapping up the 7.15 version at the moment and unfortunately these proposed changes won't be able to make it in there. 

We'd like to extend a big thank you the efforts on this so far and we would love it if you could have a look at porting this feature over to v8 instead. Of course if you have any questions on where to start then feel free to [ask on the forum](https://our.umbraco.com/forum/contributing-to-umbraco-cms/), we monitor all questions there and we're happy to assist.

Thanks again for the work in this PR and we hope to see you contributing to v8 some time soon!

Best, Sebastiaan and the Umbraco CMS team.

**Comment by jtemperv** [comment]
@ssougnez do you have intentions to port this to v8?
